src/taipy/core/_core.py

```diff
@@ -45,8 +45,8 @@ def run(self, force_restart=False):
 
         This function check the configuration, start a dispatcher and lock the Config.
         """
-        self.__check_config()
         self.__manage_version(*self.cli_args)
+        self.__check_config()
```

Comment:
This is to create the production version first before checking the config to make sure the MigrationConfig checker can detect the production version

Reply:
You're right. I'll revert the change and find another way to do this.